Lib/test/test_httpservers.py

```diff
@@ -1535,6 +1537,197 @@ def test_server_test_ipv4(self, _):
+                                          tls_cert=None,
+                                          tls_key=None,
+                                          tls_password=None)
+
+    @unittest.skipIf(ssl is None, "requires ssl")
+    def test_tls_flag(self, mock_func):
+        tls_cert_options = ['--tls-cert', ]
+        tls_key_options = ['--tls-key', ]
+        tls_password_options = ['--tls-password-file', ]
+        # Normal: --tls-cert and --tls-key
+
+        for tls_cert_option in tls_cert_options:
+            for tls_key_option in tls_key_options:
+                self.invoke_httpd(tls_cert_option, self.tls_cert,
+                                  tls_key_option, self.tls_key)
+                self.args['tls_cert'] = self.tls_cert
+                self.args['tls_key'] = self.tls_key
+                mock_func.assert_called_once_with(**self.args)
+                self.args['tls_cert'] = None
+                self.args['tls_key'] = None
+                mock_func.reset_mock()
+
+        # Normal: --tls-cert, --tls-key and --tls-password-file
+
+        for tls_cert_option in tls_cert_options:
+            for tls_key_option in tls_key_options:
+                for tls_password_option in tls_password_options:
+                    self.invoke_httpd(tls_cert_option,
+                                      self.tls_cert,
+                                      tls_key_option,
+                                      self.tls_key,
+                                      tls_password_option,
+                                      self.tls_password_file)
+                    self.args['tls_cert'] = self.tls_cert
+                    self.args['tls_key'] = self.tls_key
+                    self.args['tls_password'] = self.tls_password
+                    mock_func.assert_called_once_with(**self.args)
+                    self.args['tls_cert'] = None
+                    self.args['tls_key'] = None
+                    self.args['tls_password'] = None
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-key without --tls-cert
+
+        for tls_key_option in tls_key_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd(tls_key_option, self.tls_key)
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file without --tls-cert
+
+        for tls_password_option in tls_password_options:
+            with self.assertRaises(SystemExit):
+                self.invoke_httpd(tls_password_option, self.tls_password_file)
+                mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file cannot be opened
+
+        non_existent_file = self.get_random_temporary_file_name()
+        retry_count = 0
+        while os.path.exists(non_existent_file) and retry_count < 10:
+            non_existent_file = self.get_random_temporary_file_name()
+        if not os.path.exists(non_existent_file):
```

Comment:
This is too complicate. Just assume that some file does not exist or even use the test directory itself as the root where the file is supposed to be located. If users add some file there to make the test fail, we shouldn't care.